plugins/modules/clusters.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+from __future__ import absolute_import, division, print_function
+
+import os
+import requests
+
+
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+---
+module: clusters
+
+short_description: Handle AssistedInstall clusters
+
+version_added: "1.0.0"
+
+description: Interact with clusters through the AssistedInstall API
+
+options:
+    action:
+      description: The action to perform (list, create, delete)
+      required: false
+      default: "list"
+    cluster_id:
```

Comment:
Is this possible to make it a sub-option under query_params? There are query params such as `get _unregistered_clusters`, `openshift_cluster_id`, `ams_subscriptions_ids`, `with_hosts`, and 'owner'. THis would translate to `https://api.openshift.com/api/assisted-install/v2/clusters?openshift_cluster_id=val&get_unregistered_clysters=val&ams_subscriptions_ids=val&owner=val&with_hosts=val`. I believe this would be helpful to make the mapping with the API at https://api.openshift.com/?urls.primaryName=assisted-service%20service#/installer/v2ListClusters. Additionally, this could help with differentiating the REST API Call at https://api.openshift.com/?urls.primaryName=assisted-service%20service#/installer/ListClusterHosts. What are your thoughts?
Do we need the list action or just create, delete and update.?

Reply:
I also see other API calls such as below . How would the below be handled? Would the current approach be easily extended to the below as well.? THanks

`POST /v2/clusters/{cluster_id}/actions/complete-installation`
`GET /v2/clusters/{cluster_id}/hosts`

Reply:
> Is this possible to make it a sub-option under query_params?

Yes, it is possible to have suboptions - https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html#documentation-fields

>  Do we need the list action or just create, delete and update.?

I think `list` is good to have. Do you have any concern about it?

By using `action` I see that it makes easier the separation of tasks in the code. Adding a new one should be easy to do and to understand.


> How would the below be handled? 
> POST /v2/clusters/{cluster_id}/actions/complete-installation
>  GET /v2/clusters/{cluster_id}/hosts

It depends, Do we need them? if we do, then one way of interact with the modification, could be set through the `action`, for instance `complete-installation` could be a `action: update` or `action: complete` and only require the `cluster_id` to do so.

For the 2nd example we could avoid its implementation by pointing out that it's covered with `with_hosts` and is up to the user to access that data.

Reply:
Thanks for the response.